File: src/test/resources/fixtures/sample.java
package com.example.sample;

import java.util.List;
import java.util.ArrayList;

/**
 * A sample Java class used as a test fixture.
 */
public class SampleJavaClass {

    private final String name;

    public SampleJavaClass(String name) {
        this.name = name;
    }

    /**
     * Returns a greeting string.
     */
    public String greet() {
        return "Hello, " + name + "!";
    }

    public int compute(int x, int y) {
        return x + y;
    }

    public interface Processor {
        String process(String input);
        int count(List<String> items);
    }

    public static class NestedHelper {
        public String help() {
            return "helping";
        }
    }
}

File: src/test/resources/fixtures/sample.kt
package com.example.sample

import java.util.logging.Logger

/**
 * A sample Kotlin class used as a test fixture.
 */
class SampleKotlinClass(val name: String) {

    constructor(name: String, value: Int) : this(name) {
        println("Secondary constructor: $name=$value")
    }

    /**
     * Returns a greeting string.
     */
    fun greet(): String {
        return "Hello, $name!"
    }

    fun compute(x: Int, y: Int): Int {
        return x + y
    }

    class NestedHelper {
        fun help(): String = "helping"
    }
}

fun topLevelHelper(input: String): String {
    return input.uppercase()
}
